src/TextButton/TextButton.js

```diff
@@ -73,9 +73,11 @@ export default class WrappedTextButton extends Component {
 
   submitAction = async () => {
     let { action } = this.props
+
     this.setState({ loading: true })
-    let result = action()
-    await result
+
+    await action()
+
     this.setState({ loading: false })
   }
```

Comment:
Couldn't we just wrap the body of the action with an if statement? This would prevent the need of checking if an action exists in the `onPress` prop.

```js
submitAction = async () => {
  const { action } = this.props
  if (action && typeof action === 'function') {
    this.setState({ loading: true })
    await action()
    this.setState({ loading: false })
  }
}
```

```js
<Button
  // ...
  onPress={this.submitAction}
  // ...
/>
```